src/ripple/app/tests/TxQ_test.cpp

```diff
@@ -2398,6 +2426,270 @@ class TxQ_test : public beast::unit_test::suite
         }
     }
 
+    void testClearQueuedAccountTxs()
+    {
+        using namespace jtx;
+
+        Env env(*this,
+            makeConfig({ { "minimum_txn_in_ledger_standalone", "3" } }),
+            features(featureFeeEscalation));
+        auto alice = Account("alice");
+        auto bob = Account("bob");
+
+        checkMetrics(env, 0, boost::none, 0, 3, 256);
+        env.fund(XRP(50000000), alice, bob);
+
+        fillQueue(env, alice);
+
+        testcase("straightfoward positive case");
+        {
+            // Queue up some transactions at a too-low fee.
+            auto aliceSeq = env.seq(alice);
+            for (int i = 0; i < 2; ++i)
+            {
+                env(noop(alice), fee(100), seq(aliceSeq++), ter(terQUEUED));
+            }
+
+            // Queue up a transaction paying the open ledger fee
+            // This will be the first tx to call the operative function,
+            // but it won't succeed.
+            env(noop(alice), openLedgerFee(env), seq(aliceSeq++),
+                ter(terQUEUED));
+
+            checkMetrics(env, 3, boost::none, 4, 3, 256);
+
+            // Figure out how much it would cost to cover all the
+            // queued txs + itself
+            std::uint64_t totalFee1;
+            {
+                auto totalFactor = 0;
+                auto const metrics = env.app().getTxQ().getMetrics(
+                    env.app().config(), *env.current());
+                for (int i = 0; i < metrics->txCount + 1; ++i)
+                {
+                    auto inLedger = metrics->txInLedger + i;
+                    totalFactor += inLedger * inLedger;
+                }
+                totalFee1 = mulDiv(metrics->medFeeLevel * totalFactor /
+                    (metrics->txPerLedger * metrics->txPerLedger),
+                        env.current()->fees().base,
+                            metrics->referenceFeeLevel).second;
+                // Subtract the fees already paid
+                totalFee1 -= 100 * 2 + 8889;
+            }
```

Comment:
This `totalFee` calculation is used in several places in this file. I could be replaced with a lambda or static function. I'd prefer a lambda is nice because it can capture the `env`. A static function is fine too.